ui/src/api_tests/api.spec.ts

```diff
@@ -33,37 +41,293 @@ describe('main', () => {
+
+      // Now do beta replying to that comment, as a child comment
+      let contentBeta = 'A child federated comment from beta';
+      let commentFormBeta: CommentForm = {
+        content: contentBeta,
+        post_id: getPostRes.post.id,
+        parent_id: getPostRes.comments[0].id,
+        auth: lemmyBetaAuth,
+      };
+
+      let createResponseBeta: CommentResponse = await fetch(
+        `${lemmyBetaApiUrl}/comment`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: wrapper(commentFormBeta),
+        }
+      ).then(d => d.json());
+
+      expect(createResponseBeta.comment.content).toBe(contentBeta);
+      expect(createResponseBeta.comment.community_local).toBe(true);
+      expect(createResponseBeta.comment.creator_local).toBe(true);
+      expect(createResponseBeta.comment.parent_id).toBe(1);
+      expect(createResponseBeta.comment.score).toBe(1);
+
+      // Make sure lemmy alpha sees that new child comment from beta
+      let getPostUrlAlpha = `${lemmyAlphaApiUrl}/post?id=2`;
+      let getPostResAlpha: GetPostResponse = await fetch(getPostUrlAlpha, {
+        method: 'GET',
+      }).then(d => d.json());
+
+      // The newest show up first
+      expect(getPostResAlpha.comments[0].content).toBe(contentBeta);
+      expect(getPostResAlpha.comments[0].community_local).toBe(false);
+      expect(getPostResAlpha.comments[0].creator_local).toBe(false);
+      expect(getPostResAlpha.comments[0].score).toBe(1);
+    });
   });
 
-  function wrapper(form: any): string {
-    return JSON.stringify(form);
-  }
+  describe('update test comment', () => {
+    test('/u/lemmy_alpha updates a comment on /c/lemmy_beta/main, its on both instances', async () => {
+      let content = 'A jest test federated comment update';
+      let commentForm: CommentForm = {
+        content,
+        post_id: 2,
+        edit_id: 1,
+        auth: lemmyAlphaAuth,
+        creator_id: 2,
+      };
+
+      let updateResponse: CommentResponse = await fetch(
+        `${lemmyAlphaApiUrl}/comment`,
+        {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: wrapper(commentForm),
+        }
+      ).then(d => d.json());
+
+      expect(updateResponse.comment.content).toBe(content);
+      expect(updateResponse.comment.community_local).toBe(false);
+      expect(updateResponse.comment.creator_local).toBe(true);
+
+      let getPostUrl = `${lemmyBetaApiUrl}/post?id=2`;
+      let getPostRes: GetPostResponse = await fetch(getPostUrl, {
+        method: 'GET',
+      }).then(d => d.json());
+
+      expect(getPostRes.comments[1].content).toBe(content);
+      expect(getPostRes.comments[1].community_local).toBe(true);
+      expect(getPostRes.comments[1].creator_local).toBe(false);
+    });
+  });
 });
+
+function wrapper(form: any): string {
+  return JSON.stringify(form);
+}
```

Comment:
Maybe we should split this into seperate files, and extract some helper functions. But again, not necessarily in this PR.

Reply:
We could even remove it, I thought it was needed to wrap it in something like `data: {...` but it wasn't, just the stringified form works for http reqs, unlike websocket which needs a wrapper with the operation type.